### ISSUE TITLE
Skip cmp-report if there is no JAR?

### ISSUE THREAD

**Issue by stain** [siom79/japicmp#148]
In Apache Commons we've tried using japicmp report, which works great if you do `mvn clean verify site`, but fails if you only do `mvn clean site`

> [debug] Searching for versions in versionRange: (,1.9.3)
> [debug] Parameter <oldVersionPattern> not configured, i.e. no version filtered.
> [warn] No new version specified and file '/home/stain/src/rc2/91/commons-beanutils-1.9.3-src/target/classes' of artifact could not be opened as jar archive: /home/stain/src/rc2/91/commons-beanutils-1.9.3-src/target/classes (Is a directory)

I guess this is because it can't find the JAR file, which is made in the `package` phase. 

We have japicmp [listed in the `<reporting>` section](http://central.maven.org/maven2/org/apache/commons/commons-parent/41/commons-parent-41.pom).

Is there a way to conditionally skip the japicmp report if `package` has not been executed (e.g. there is no new version)? E.g. a configuration

````xml
<parameter>
   <ignoreMissingNewVersion>true</ignoreMissingNewVersion>
</parameter>
```

**Comment by siom79** [comment]
Currently there is no option to disable the report if no archive is available for comparison. I will investigate if it is possible to detect in the plugin if the phase package has been run or not. Otherwise the plugin can skip its execution, but then you also don't have a report...

**Comment by stain** [comment]
Thanks - yeah, that's why I think it should be configured as otherwise it could just be that the `<newVersion>` was wrong.

Perhaps just log something like

> [WARNING] Could not find newVersion target/blablah.jar - japicmp report skipped.

**Comment by siom79** [comment]
Added new paramater `<ignoreMissingNewVersion/>` on the `develop` branch.

If you like to try out:
```
git clone https://github.com/siom79/japicmp.git
cd japicmp
git checkout develop
mvn install
```

It will be released with 0.9.1.

**Comment by garydgregory** [comment]
OK, great. When can we expect a release?

**Comment by siom79** [comment]
I hope I find some time this week or at least next weekend.

**Comment by siom79** [comment]
Released 0.9.1.

**Comment by stain** [comment]
Thanks! Works well.
